packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/ModelTypeProvider.cs

```diff
@@ -177,31 +104,26 @@ private IReadOnlyList<Parameter> BuildConstructorParameters(bool isSerialization
             foreach (var property in _inputModel.Properties)
             {
                 CSharpType propertyType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(property.Type);
-                var initializationValue = GetPropertyInitializationValue(property, propertyType);
-                var parameterValidation = GetParameterValidation(property, propertyType);
-
-                var parameter = new Parameter(
-                   Name: property.Name.ToVariableName(),
-                   Description: FormattableStringHelpers.FromString(property.Description),
-                   Type: propertyType,
-                   DefaultValue: null,
-                   Validation: parameterValidation,
-                   Initializer: null);
-
                 // All properties should be included in the serialization ctor
                 if (isSerializationConstructor)
                 {
-                    constructorParameters.Add(parameter with { Validation = ValidationType.None });
+                    constructorParameters.Add(new Parameter(property)
+                    {
+                        Validation = ValidationType.None,
```

Comment:
isn't this the default?

Reply:
It is the default value, if it's not set. When a parameter is constructed from either an `InputModelProperty` or `InputParameter` we are setting the validation based on some checks on the input and it's type. In this case here, we overwrite it since we want no validation.

Reply:
Can you open an issue to track moving validation from a parameter property to a method body consideration?  I think this is an artifact of the old days where the writer was responsible for all method body statements so the writer needed to know this information for every param.  Now that a method consists of a signature and a body the writer no longer needs to know this information.  This can be done in a follow up PR.

Reply:
opened https://github.com/microsoft/typespec/issues/3486.